python/ray/data/tests/test_dataset.py

```diff
@@ -5085,6 +5085,19 @@ def test_default_batch_format(shutdown_only):
     assert ds.default_batch_format() == pd.DataFrame
 
 
+def test_dataset_schema_after_read_stats(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(num_cpus=1)
+    ray.init(cluster.address)
+    cluster.add_node(num_cpus=1, resources={"foo": 1})
+    ds = ray.data.read_csv(
+        "example://iris.csv", ray_remote_args={"resources": {"foo": 1}}
+    )
+    schema = ds.schema()
+    ds.stats()
+    assert schema == ds.schema()
```

Comment:
What does this test require using cluster utils and custom resources to reproduce? 🤔

Reply:
It's to make stats actor record task in a different task:
```
def test_local_stats_actor(ray_start_cluster):
    from ray.data._internal.stats import _get_or_create_stats_actor
    cluster = ray_start_cluster
    cluster.add_node(num_cpus=1)
    ray.init(cluster.address)
    ac = _get_or_create_stats_actor()
    ac.record_start.remote(123)

    @ray.remote
    def rtask(ac):
        meta = BlockMetadata(num_rows=1, size_bytes=8, schema=None, input_files=None, exec_stats=None)
        ac.record_task.remote(123, 0, [meta])

    rtask.remote(ac)

    print(ray.get(ac.get.remote(123)))
    # Output empty instead of the desired entry: ({}, 0.0)
```